src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -435,15 +482,17 @@ internal async Task<MigrationAssessmentResult> GetAssessmentItems(string[] conne
             SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath = Path.GetDirectoryName(Logger.LogFileFullPath);
             DmaEngine engine = new DmaEngine(connectionStrings);
             ISqlMigrationAssessmentModel contextualizedAssessmentResult = await engine.GetTargetAssessmentResultsListWithCheck(System.Threading.CancellationToken.None);
-            engine.SaveAssessmentResultsToJson(contextualizedAssessmentResult, false);
+            var assessmentFileName = String.Format("SqlAssessmentReport-{0}.json", DateTime.UtcNow.ToString("yyyyMMddHHmmss"));
+            engine.SaveAssessmentResultsToJson(contextualizedAssessmentResult, false, Path.Combine(SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath, assessmentFileName));
```

Comment:
so wait, we offer two separate functionalities for SKU rec - `GetSkuRecommendationsRequest` to generate the actual recommendation, and then `SaveSkuRecommendationsResultRequest` to save the report. Whereas for assessment we only offer one functionality, `MigrationAssessmentsRequest` which both generates the actual assessment and also saves it all in one. Why is SKU rec split into two calls but assessment is only one? Should it not be consistent for both?  i.e. we either offer a dedicated save report button for both SKU rec and assessment, or we build the save report functionality directly into the call that actually generates the results

Reply:
good catch. I think the previous behavior is automatically saving the assessment result in default folder. now we can remove this line.